numpyro/infer/hmc_util.py

```diff
@@ -431,7 +432,10 @@ def update_fn(t, accept_prob, z_info, state):
         :return: new state of the adapt scheme.
         """
         step_size, inverse_mass_matrix, mass_matrix_sqrt, ss_state, mm_state, window_idx, rng_key = state
-        rng_key, rng_key_ss = random.split(rng_key)
+        if rng_key is not None:
```

Comment:
why was this necessary?

Reply:
We don't use `find_reasonable_step_size` stuff, so `adapt_state.rng_key` is set to `None` in Barker.